src/Jackalope/Transport/DoctrineDBAL/Client.php

```diff
@@ -2358,8 +2381,22 @@ public function getSupportedQueryLanguages()
      */
     public function registerNamespace($prefix, $uri)
     {
-        if (isset($this->namespaces[$prefix]) && $this->namespaces[$prefix] === $uri) {
-            return;
+        if (isset($this->namespaces[$prefix])) {
+            if ($this->namespaces[$prefix] === $uri) {
+                return;
+            }
+
+            if (isset($this->coreNamespaces[$prefix])) {
+                throw new NamespaceException("Cannot overwrite JCR core namespace prefix '$prefix' to a new uri '$uri'.");
+            }
+        }
+
+        if (!$this->namespaces instanceof \ArrayObject) {
```

Comment:
we have the same 2 if statements in ``unregisterNamespace()`` .. not sure though if it makes sense to move that into a separate method.